app/process/handler/consensus-message/consensus-message.go

```diff
@@ -82,7 +91,32 @@ func (cmh ConsensusMessageHandler) Handle(payload []byte) {
 }
 
 func (cmh ConsensusMessageHandler) handleEthTxMessage(m *validatorproto.TopicEthTransactionMessage) error {
-	// TODO: verify authenticity of transaction hash
+	err := cmh.transactionRepository.UpdateStatusEthTxSubmitted(m.TransactionId, m.EthTxHash)
+	if err != nil {
+		cmh.logger.Errorf("Failed to update status to [%s] of transaction with TransactionID [%s]. Error [%s].", transaction.StatusEthTxSubmitted, m.TransactionId, err)
+		return err
+	}
+
+	isSuccessful, err := cmh.ethereumClient.WaitForTransactionSuccess(common.HexToHash(m.EthTxHash))
+	if err != nil {
+		cmh.logger.Errorf("Failed await a transaction with Id [%s] and Hash [%s]. Error [%s].", m.TransactionId, m.Hash, err)
```

Comment:
`Failed to await TX ID [%s] with ETH TX [%s] to be mined. Error [%s].`, m.TransactionID, m.EthHash, err